pkg/controller/packetcapture/packetcapture_controller.go

```diff
@@ -0,0 +1,338 @@
+	provider            operatorv1.Provider
+	enterpriseCRDsExist bool
+	status              status.StatusManager
+	clusterDomain       string
+	usePSP              bool
+	licenseAPIReady     *utils.ReadyFlag
+	tierWatchReady      *utils.ReadyFlag
+	multiTenant         bool
+}
+
+// Reconcile reads that state of the cluster for a PacketCapture object and makes changes based on the state read
+// and what is in the PacketCapture.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcilePacketCapture) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling PacketCapture")
+
+	packetcaptureapi, err := utils.GetPacketCaptureAPI(ctx, r.client)
+	if err != nil && !errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying PacketCapture", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	r.status.OnCRFound()
+	reqLogger.V(2).Info("Loaded config", "config", packetcaptureapi)
+
+	defer r.status.SetMetaData(&packetcaptureapi.ObjectMeta)
+
+	// Changes for updating PacketCapture status conditions.
+	if request.Name == ResourceName && request.Namespace == "" {
+		ts := &operatorv1.TigeraStatus{}
+		if err := r.client.Get(ctx, types.NamespacedName{Name: ResourceName}, ts); err != nil {
+			return reconcile.Result{}, err
+		}
+		packetcaptureapi.Status.Conditions = status.UpdateStatusCondition(packetcaptureapi.Status.Conditions, ts.Status.Conditions)
+		if err := r.client.Status().Update(ctx, packetcaptureapi); err != nil {
+			log.WithValues("reason", err).Info("Failed to create packetcapture status conditions.")
+			return reconcile.Result{}, err
+		}
+	}
+
+	variant, installationSpec, err := utils.GetInstallation(context.Background(), r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceNotFound, "Installation not found", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying installation", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	if variant != operatorv1.TigeraSecureEnterprise {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise), nil, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	managementCluster, err := utils.GetManagementCluster(ctx, r.client)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading ManagementCluster", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Packet capture is disabled in multi tenant management cluster
+	if r.multiTenant && managementCluster != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
Should i set the status degraded if we encounter "packet capture" in multi-tenant management cluster? @caseydavenport

Reply:
Updated the code: The packet capture  would be in a degraded status if it is found inside a multitenant management cluster.